File: Serene/Serene.Script/Imports/ServiceContracts/Northwind.SupplierRow.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class SupplierRow
    {
        [InlineConstant] public const string IdProperty = "SupplierID";
        [InlineConstant] public const string NameProperty = "CompanyName";
        [InlineConstant] public const string LocalTextPrefix = "Northwind.Supplier";
        [InlineConstant] public const string LookupKey = "Northwind.Supplier";
    
        public static Lookup<SupplierRow> Lookup { [InlineCode("Q.getLookup('Northwind.Supplier')")] get { return null; } }
    
        public Int32? SupplierID { get; set; }
        public String CompanyName { get; set; }
        public String ContactName { get; set; }
        public String ContactTitle { get; set; }
        public String Address { get; set; }
        public String City { get; set; }
        public String Region { get; set; }
        public String PostalCode { get; set; }
        public String Country { get; set; }
        public String Phone { get; set; }
        public String Fax { get; set; }
        public String HomePage { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string SupplierID = "SupplierID";
            [InlineConstant] public const string CompanyName = "CompanyName";
            [InlineConstant] public const string ContactName = "ContactName";
            [InlineConstant] public const string ContactTitle = "ContactTitle";
            [InlineConstant] public const string Address = "Address";
            [InlineConstant] public const string City = "City";
            [InlineConstant] public const string Region = "Region";
            [InlineConstant] public const string PostalCode = "PostalCode";
            [InlineConstant] public const string Country = "Country";
            [InlineConstant] public const string Phone = "Phone";
            [InlineConstant] public const string Fax = "Fax";
            [InlineConstant] public const string HomePage = "HomePage";
        }
    }
    
}

File: Serene/Serene.Core/Imports/ServerTypings/Northwind.SupplierRow.ts
namespace Serene.Northwind {
    export interface SupplierRow {
        SupplierID?: number;
        CompanyName?: string;
        ContactName?: string;
        ContactTitle?: string;
        Address?: string;
        City?: string;
        Region?: string;
        PostalCode?: string;
        Country?: string;
        Phone?: string;
        Fax?: string;
        HomePage?: string;
    }

    export namespace SupplierRow {
        export const idProperty = 'SupplierID';
        export const nameProperty = 'CompanyName';
        export const localTextPrefix = 'Northwind.Supplier';
        export const lookupKey = 'Northwind.Supplier';

        export function getLookup(): Q.Lookup<SupplierRow> {
            return Q.getLookup<SupplierRow>('Northwind.Supplier');
        }

        export namespace Fields {
            export declare const SupplierID: string;
            export declare const CompanyName: string;
            export declare const ContactName: string;
            export declare const ContactTitle: string;
            export declare const Address: string;
            export declare const City: string;
            export declare const Region: string;
            export declare const PostalCode: string;
            export declare const Country: string;
            export declare const Phone: string;
            export declare const Fax: string;
            export declare const HomePage: string;
        }

        [
            'SupplierID', 
            'CompanyName', 
            'ContactName', 
            'ContactTitle', 
            'Address', 
            'City', 
            'Region', 
            'PostalCode', 
            'Country', 
            'Phone', 
            'Fax', 
            'HomePage'
        ].forEach(x => (<any>Fields)[x] = x);
    }
}
